File: core/src/net/natruid/jungle/components/render/ScaleComponent.java
package net.natruid.jungle.components.render;

public class ScaleComponent extends Vector2Component {
    public static final ScaleComponent DEFAULT = new ScaleComponent();

    public ScaleComponent() {
        super(1, 1);
    }

    public ScaleComponent(float x, float y) {
        super(x, y);
    }

    @Override
    protected void reset() {
        set(DEFAULT);
    }
}

File: core/src/net/natruid/jungle/components/render/ScaleComponent.kt
package net.natruid.jungle.components.render

class ScaleComponent(x: Float = 1f, y: Float = 1f) : Vector2Component(x, y) {
    companion object {
        val DEFAULT = ScaleComponent()
    }

    override fun reset() {
        set(1f, 1f)
    }
}
